core/src/HIP/Kokkos_HIP_Space.hpp

```diff
@@ -76,12 +84,12 @@ class HIPSpace {
                  const size_t arg_logical_size = 0) const {
     return allocate(arg_label, arg_alloc_size, arg_logical_size);
   }
-#else
+#endif
```

Comment:
Please confirm that this change is intentional

Reply:
Yes, we always need the member function below to allocate on the correct device. The ones abover are only necessary for unified memory.